Source/CarthageKit/Cartfile.swift

```diff
@@ -92,6 +92,27 @@ extension Cartfile: Printable {
 	}
 }
 
+// Duplicate dependencies
+extension Cartfile {
+	/// Returns an array containing projects that are listed as duplicate
+	/// dependencies.
+	public func duplicateProjects() -> [ProjectIdentifier] {
+		let projectSet = buildCountedSet(self.dependencies.map { $0.project })
```

Comment:
This bit is duplicated below. Could it be factored out into a `private` function and reused by both?